pkg/operator/ceph/controller/spec.go

```diff
@@ -653,8 +655,27 @@ func PrivilegedContext(runAsRoot bool) *v1.SecurityContext {
 	return sec
 }
 
-// LogCollectorContainer runs a cron job to rotate logs
+// LogCollectorContainer rotate logs
 func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Container {
+
+	maxLogSize := "500M"
+	if c.LogCollector.MaxLogSize != nil {
+		maxLogSize = fmt.Sprintf("%dM", c.LogCollector.MaxLogSize.Value()/1000/1000)
+		if maxLogSize == "0M" {
```

Comment:
Use `resource.IsZero()` instead?

Reply:
since we have to work resources in the string and not the` resource.Quantity` type. so keeping it as it is.

Reply:
`IsZero()` is still cleaner IMO and you can call `String()` to transform once you need it.

Reply:
I'm happy to make the change but we are converting the resource to MB here `c.LogCollector.MaxLogSize.Value()/1000/1000` which is `int64` and after this conversation, we check if the logSize that we are getting is `OM` or not.  

I tried 
```
maxLogSize := resource.NewQuantity(500000000, resource.DecimalSI)
	if c.LogCollector.MaxLogSize != nil {
		maxLogSize = c.LogCollector.MaxLogSize.ToDec()
		if maxLogSize.IsZero() {
			logger.Info("maxLogSize is 0M setting to minimum of 1M")
			maxLogSize = resource.NewQuantity(1000000, resource.DecimalSI)
		}
	}

****
			fmt.Sprintf(cronLogRotate, daemonID, periodicity, maxLogSize.String()),
```
but this line `		maxLogSize = c.LogCollector.MaxLogSize.ToDec()`  will have problem when user give input in this format `Mi/MiB` since logrotate  will take `M` as valid input(for example)

 let me know if I'm missing something.

Reply:
What about this comparison for 0? It is nice to compare numerically when possible.
```
		size := c.LogCollector.MaxLogSize.Value() / 1000 / 1000
		if size == 0 {
			size = 1
			logger.Info("maxLogSize is 0M setting to minimum of 1M")
		}
		maxLogSize = fmt.Sprintf("%dM", size)
```